pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java

```diff
@@ -112,6 +113,13 @@ public PinEntryEditText(Context context, AttributeSet attrs, int defStyleAttr, i
         init(context, attrs);
     }
 
+    public void setMaxLength(final int maxLength) {
+        mMaxLength = maxLength;
+        mNumChars = maxLength;
+
+        setFilters(new InputFilter[]{new InputFilter.LengthFilter(maxLength)});
```

Comment:
you probably want to call invalidate() after this.
Because if the widget is displaying already when the max filter is changed, the change will not be reflected
until the screen is redrawn. I've looked at the code for setFilters and doesn't look like it calls invalidate at any point (understandable since there is no visual change normally). You also want to clear any values currently in the widget otherwise it may result in unexpected behaviour. For example, the widget holds 5 out of 6 characters, then the code changes it to hold only 4 characters... the outcome is an unpredicted state.